lib/exercises.rb

```diff
@@ -2,22 +2,67 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n*m) (looping through arrays and word length)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  final_arrays = {}
+  
+  strings.each do |string|
+    letters_count = {}
+    string.each_char do |char|
+      if letters_count[char]
+        letters_count[char] += 1
+      else
+        letters_count[char] = 1
+      end
+    end
+    
+    if final_arrays[letters_count]
+      final_arrays[letters_count] << string
+    else
+      final_arrays[letters_count] = [string]
+    end
+  end
+  
+  output = []
+  final_arrays.each do |k, v| 
+    output << v
+  end
+  return output
+  
 end
 
+
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
Your runtime complexity is actually O(nk) see below for why.  This works, but it's not quite optimal.